gzip.go

```diff
@@ -88,7 +88,9 @@ func (grw gzipResponseWriter) Write(p []byte) (int, error) {
 	if len(grw.Header().Get(HeaderContentType)) == 0 {
 		grw.Header().Set(HeaderContentType, http.DetectContentType(p))
 	}
-
+	// removing content length before writing in case gzip.Writer writes to the
+	// underlying http.ResponseWriter before closing. See notes in compress/gzip.
```

Comment:
Here is the [link](https://golang.org/pkg/compress/gzip/#Writer.Write) to compress/gzip.

Extracted comment:
Write writes a compressed form of p to the underlying io.Writer. The compressed bytes are not necessarily flushed until the Writer is closed.